contracts/OxoFactory.sol

```diff
@@ -0,0 +1,173 @@
+pragma solidity ^0.4.15;
+
+contract OxoFactory {
+
+    mapping (address => uint) public userIdByAddr;
+    mapping (address => User) public userByAddr;
+    mapping (uint => Game) public gameById;
+
+	event NewUserEvent(string _text , string _newUser);
+	event NewGameEvent (string _text , uint _id);
+    event EventMessage(string _text);
+
+	enum  State { WaitForPlayers , WaitPlayerOne, WaitPlayerTwo, ReadyToPlay, WaitStepFirstPlayer, WaitStepSecondPlayer ,WinnerPlayerOne, WinnerPlayerTwo, Draw}
+
+	struct User{
+	    string name;
+	    uint gameId;
+	}
+
+    struct Game{
+		address playerOne;
+		address playerTwo;
+		uint[] board;
+		State state;
+		uint8 countPlayers;
+		address next;
+		uint8 countOfMove;
+		string status;
+		uint money;
+	}
+
+	Game[] games;
+
+	function register(string _name) public {
+        require(userIdByAddr[msg.sender] == 0);
+        require(keccak256(_name) != keccak256("") && keccak256(_name) != keccak256(" "));
+        userIdByAddr[msg.sender] = uint(keccak256(_name));
+        User memory newUser;
+        newUser.name = _name;
+        userByAddr[msg.sender] = newUser;
+		NewUserEvent( "New User was register", _name );
+    }
+
+	 function createNewGame() payable public onlyKnownUser onlyFinishGame {
+		 Game memory someGame;
+		 someGame.state = State.WaitForPlayers;
+		 someGame.status = stateToString(State.WaitForPlayers);
+		 uint[] memory array = new uint[](9);
+		 	for (uint i = 0; i < array.length; i++)
+                   array[i] =  uint(keccak256("empty"));
+		 someGame.board =array;
+		 uint id = games.push(someGame) - 1;
+		 someGame.money = msg.value;
+		 gameById[id] = someGame;
+		 userByAddr[msg.sender].gameId = id;
+		 NewGameEvent("New Game was created", id );
+    }
+
+    function joinToGameById(uint _id) payable public onlyKnownUser onlyFinishGame returns(bool){
```

Comment:
what will happen if I try to join to game that wasn't created? Do you have a test for this case?

Reply:
yeap. You are right.
I have added verification and several tests for it. Please look again.